File: 1186-maximum-subarray-sum-with-one-deletion/1186-maximum-subarray-sum-with-one-deletion.cpp
class Solution {
public:
    int maximumSum(vector<int>& arr) {

        int n = arr.size();

        int noDelete = arr[0];
        int oneDelete = INT_MIN;

        int ans = arr[0];

        for (int i = 1; i < n; i++) {

            int prevNoDelete = noDelete;

            noDelete = max(noDelete + arr[i], arr[i]);

            if (oneDelete == INT_MIN) {
                oneDelete = prevNoDelete;
            }
            else {
                oneDelete = max(oneDelete + arr[i], prevNoDelete);
            }

            ans = max(ans, max(noDelete, oneDelete));
        }

        return ans;
    }
};

File: 1186-maximum-subarray-sum-with-one-deletion/1186-maximum-subarray-sum-with-one-deletion.py
class Solution:
    def maximumSum(self, arr: List[int]) -> int:
        n = len(arr)

        ans = arr[0]
        noDelete = arr[0]
        oneDelete = float('-inf')

        for i in range(1,n):

            prevNoDelete = noDelete

            noDelete = max(noDelete + arr[i],arr[i])

            if oneDelete == float('-inf'):
                oneDelete = prevNoDelete
            else:
                oneDelete = max(oneDelete+arr[i],prevNoDelete)
            
            ans = max(ans,max(oneDelete,noDelete))
        return ans 